sql-common/my_time.c

```diff
@@ -336,12 +339,25 @@ str_to_datetime(const char *str, uint length, MYSQL_TIME *l_time,
     pos++;
     skip_digits(&pos, end); // ignore the return value
   }
+  if (pos < end && *pos == '-' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu-HH:MM is supported too */
```

Comment:
for readability I'd merge with the next branch as its the same condition
<code>
if (pos < end && (*pos == '-' || *pos == '+') && digits >= 12)
</code>
Likewise for later too.